app/hooks/keyboardAnimation.ts

```diff
@@ -0,0 +1,276 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useKeyboardHandler} from 'react-native-keyboard-controller';
+import {useAnimatedScrollHandler, useDerivedValue, useSharedValue} from 'react-native-reanimated';
+
+import {BOTTOM_TAB_HEIGHT} from '@constants/view';
+
+export const useKeyboardAnimation = (
+    postInputContainerHeight: number,
+    enableAnimation = true,
+    isTablet = false,
+    safeAreaBottom = 0,
+    isThreadView = false,
+    enabled = true,
+) => {
+    /**
+   * progress: Keyboard animation progress (0 = closed, 1 = fully open)
+   * Used for: Tracking keyboard animation state
+   */
+    const progress = useSharedValue(0);
+
+    /**
+   * height: Keyboard height (adjusted for tab bar) used to animate input container position
+   *
+   * How it works: This value represents the keyboard height and is used with negative translateY
+   * in useKeyboardAwarePostDraft.ts (transform: [{translateY: -height.value}]) to move the input
+   * container UP by the keyboard height amount. Higher keyboard height = input moves up more.
+   *
+   * Smoothed during interactive gestures to prevent jerky movements
+   */
+    const height = useSharedValue(0);
+
+    /**
+   * inset: Bottom inset for the scroll view
+   * Adds padding at the bottom of scroll content so it doesn't hide behind keyboard
+   */
+    const inset = useSharedValue(0);
+
+    /**
+   * offset: Scroll offset adjustment when keyboard opens
+   * Ensures the scroll view scrolls to the right position when keyboard appears
+   */
+    const offset = useSharedValue(0);
+
+    /**
+   * scroll: Tracks the current scroll position of the ScrollView
+   * Used to calculate proper offset when keyboard opens
+   */
+    const scroll = useSharedValue(0);
```

Comment:
I struggle to see the difference with the offset

Reply:
- offset: The keyboard height (how much to adjust by)
- scroll: he current scroll position (where we currently are)

Reply:
Change the name to `scrollPosition` for more clarity.